cmd/signature-v2.go

```diff
@@ -282,7 +282,13 @@ func canonicalizedResourceV2(encodedPath string, encodedQuery string) string {
 			canonicalQueries = append(canonicalQueries, key)
 			continue
 		}
-		canonicalQueries = append(canonicalQueries, key+"="+val)
+		// Resources values should be unescaped
+		unescapedVal, err := url.QueryUnescape(val)
+		if err != nil {
+			errorIf(err, "Unable to unescape query value (query = `%s`, value = `%s`)", key, val)
+			continue
+		}
+		canonicalQueries = append(canonicalQueries, key+"="+unescapedVal)
```

Comment:
@krishnasrinivas,

```
#!/bin/bash
 
amzFile="file"
bucket="vadmeste"
resource="/${bucket}/${amzFile}"
contentType="application/octet-stream"
dateValue=`TZ=GMT date -R | sed 's/+0000/GMT/'`
stringToSign="GET\n\n${contentType}\n${dateValue}\n${resource}?uploadId=2b10 b895"

s3Key="xxx"
s3Secret="xxx"

signature=`echo -en ${stringToSign} | openssl sha1 -hmac ${s3Secret} -binary | base64`

curl  -H "Host: ${bucket}.s3.amazonaws.com" \
-H "Date: ${dateValue}" \
-H "Content-Type: ${contentType}" \
-H "Authorization: AWS ${s3Key}:${signature}" \
     "https://${bucket}.s3.amazonaws.com/${amzFile}?max-parts=1000&part-number-marker=0&uploadId=2b10%20b895"                                                                                                                      
```

@krishnasrinivas, the above script lists parts of the upload id = "2b10 b895", notice the space inside the upload id, also notice that space (not encoded) in stringToSign. This scripts returns NoSuchUpload and not Signature mismatched.